### PR TITLE
Update French translation

### PR DESCRIPTION
**Pull request by Nicryc**
- Added remaining missing translations
- Corrected some translations
- Replaced some characters by better Unicode alternatives such as:
  - ... → … (U+2026)
  - A → À (U+00C0)
  - " :" → " :" (non-breaking space: U+00A0)
  - \- → ‑ (non-Breaking Hyphen: U+2011)

### CODE REVIEW

**Comment by laurent22** [comment]
I'm not sure we need the space changes. Do you have an example of string in the UI that's currently broken, and that would be fixed by your changes?

**Comment by Nicryc** [comment]
The UI isn't really broken but usually in French some punctuation sign (including ":", "!" or "?") are preceded by a non-breaking space. See [here](https://unicode.org/udhr/n/notes_fra.html). This is to prevent a line to begin with ":".

Joplin screenshot showing a line break between a word and ":"

As you can see in the orange frame, the "documentation :" has a line break in between resulting in the ":" being at the beginning of the new line. 

I agree it is a detail and most modifications won't be affected since they are short sentences or single words which will required an excessively little window to happen (but it might happens on mobile phone still).

Anyway if you think these modifications aren't necessary I could remove them.

By the way I noticed I made a mistake lines 490 and 491 by adding a non-breaking hyphen to a command example which shouldn't be.

File: packages/tools/locales/fr_FR.po [hunk 1]
```diff
@@ -484,9 +486,9 @@ msgid ""
 "for to-dos, or `nt` for notes and to-dos (eg. `-tt` would display only the "
 "to-dos, while `-tnt` would display notes and to-dos."
 msgstr ""
-"Affiche uniquement les notes du ou des types spécifiés. Le type peut-être "
-"`n` pour les notes, `t` pour les tâches (par exemple, `-tt` affiche "
-"uniquement les tâches, tandis que `-tnt` affiche les notes et les tâches)."
+"Affiche uniquement les notes du ou des types spécifiés. Le type peut‑être "
+"`n` pour les notes, `t` pour les tâches (par exemple, `‑tt` affiche "
+"uniquement les tâches, tandis que `‑tnt` affiche les notes et les tâches)."
```
Comment (laurent22) [inline packages/tools/locales/fr_FR.po:491]: Please don't change flags as it means if a user copy and paste them they won't work.

**Review by laurent22 — COMMENTED** [review]
Thanks, the update looks very good other than the comment I've left. Please also have a quick check to see if you didn't change any other flags.

**Comment by laurent22** [comment]
> The UI isn't really broken but usually in French some punctuation sign (including ":", "!" or "?") are preceded by a non-breaking space. See [here](https://unicode.org/udhr/n/notes_fra.html). This is to prevent a line to begin with ":".

Thanks for the detailed explanation. Your space changes do make sense and improve the UI so let's keep them.

File: packages/tools/locales/fr_FR.po (hunk 1 above)
Comment (Nicryc) [reply packages/tools/locales/fr_FR.po]: I noticed that mistake, this is now corrected. I check the whole file; no other flags have been changed.

**Comment by laurent22** [comment]
All good now, many thanks for the update!

**Pull request merged**